subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java

```diff
@@ -101,11 +105,20 @@ public DefaultSettings(
         this.settingsScript = settingsScript;
         this.startParameter = startParameter;
         this.services = serviceRegistryFactory.createFor(this);
-        this.rootProjectDescriptor = createProjectDescriptor(null, settingsDir.getName(), settingsDir);
+        this.rootProjectDescriptor = createProjectDescriptor(null, getProjectName(settingsDir), settingsDir);
         this.dependencyResolutionManagement = services.get(DependencyResolutionManagementInternal.class);
         this.toolchainManagement = services.get(ToolchainManagementInternal.class);
     }
 
+    private static String getProjectName(File settingsDir) {
+        if(contains(File.listRoots(), settingsDir)) {
+            String rootIndicator = settingsDir.toPath().getRoot().toString().replaceAll("[\\\\:\\/]*", "");
+            return "generated " + rootIndicator  + (rootIndicator.isEmpty() ? "" : " ") +
```

Comment:
I would use `-` instead of spaces as the later is going to make it harder to use the name on the CLI. Not that it should happen often.